core/src/main/java/io/cloudevents/core/extensions/DistributedTracingExtension.java

```diff
@@ -49,23 +51,31 @@ public void setTracestate(String tracestate) {
     }
 
     @Override
-    public void readFromEvent(CloudEvent event) {
-        Object tp = event.getExtension(TRACEPARENT);
+    public void readFrom(CloudEventExtensions extensions) {
+        Object tp = extensions.getExtension(TRACEPARENT);
         if (tp != null) {
             this.traceparent = tp.toString();
         }
-        Object ts = event.getExtension(TRACESTATE);
+        Object ts = extensions.getExtension(TRACESTATE);
         if (ts != null) {
             this.tracestate = ts.toString();
         }
     }
 
     @Override
-    public Map<String, Object> asMap() {
-        HashMap<String, Object> map = new HashMap<>();
-        map.put(TRACEPARENT, this.traceparent);
-        map.put(TRACESTATE, this.tracestate);
-        return Collections.unmodifiableMap(map);
+    public Object getValue(String extensionName) {
+        switch (extensionName) {
+            case TRACEPARENT:
+                return this.traceparent;
+            case TRACESTATE:
+                return this.tracestate;
+        }
+        return null;
```

Comment:
Just thinking out loud here:
Perhaps we should be more defensive and throw an error on unknown key. Should help debugging issues when wrong key is used, for example

Reply:
In that case, should we bring it to the interface? aka:

```
Object getValue(String key) throws IllegalArgumentException;
```

Reply:
Won't hurt, I suppose 👍

Reply:
like https://github.com/cloudevents/sdk-java/pull/158/commits/e2de52c46989b47aefa7c33f2b5ae91d733add57